src/test/test_elf_loader.cpp

```
@@ -151,17 +153,91 @@ SectionHeaderInfo get_section_header_info(const std::filesystem::path& path, con
+    }
     // e_machine field in ELF header (both 32-bit and 64-bit).
     write_u16_le(bytes, offsetof(ELFIO::Elf32_Ehdr, e_machine), machine);
     write_file_bytes(path, bytes);
 }
 
+void patch_section_table_offset(const std::filesystem::path& path, const uint64_t offset) {
+    auto bytes = read_file_bytes(path);
+    if (bytes.size() <= ELFIO::EI_CLASS) {
+        throw std::runtime_error("ELF header is truncated");
+    }
+    if (bytes[ELFIO::EI_CLASS] == ELFIO::ELFCLASS32) {
```

Clang (14.0.6)

[warning] repeated branch in conditional chain

(bugprone-branch-clone)